### ISSUE TITLE
Blocks editor is broken on Chrome in versions > 45 (urgent)

### ISSUE THREAD

**Issue by halatmit** [mit-cml/appinventor-sources#533]
App Inventor works OK on Chrome 45.0.2454.46 beta (64-bit)
But not on versions of 45 (not sure where) and on Version 46.0.2486.0 dev (64-bit)

When you open the Blocks editor, you don't see any blocks, and even the column of all blocks at the left of the screen is blank.

I took a look with the debugger and I think the bug message comes up in the console as 

blockly-7663623d-4c1e-4c85-8f39-44aed147555f.js:3137 Uncaught SyntaxError: Unexpected identifier(anonymous function) @ blockly-7663623d-4c1e-4c85-8f39-44aed147555f.js:3137

The reference looks like it pointing to the definition or AllUpgradeMaps in versioning.java.
My guess it that there is a typo in the upgrade table that's causing a syntax error in the generated code, which could be ugly to find.

we need to nail this quickly.  It's already breaking App Inventor on some Chromebooks and will get worse and Google pushes Chrome updates.

**Comment by halatmit** [comment]
I did a little more hunting around.  Now I'm let certain my guess above is correct.  I don't know what's going on.  Need some Javascript help.

**Comment by cpseager** [comment]
I suspect this translation has the double quote and single quote around the wrong way, JSON wants double quote around a string. We've had similar issues in other Blockly apps, Blockly gives up completely on this type of error.

Blockly.Msg.LANG_TEXT_APPEND_TOOLTIP='Añade texto a una variable "%1".';

**Comment by halatmit** [comment]
@cpseager Thanks for the tip.  We'll take a look,  But note this error does not happen  in Chrome version 44 and below.

**Comment by cpseager** [comment]
Indeed, Chrome 44 is fine for me, but Canary 47 crashes on that Spanish translation line:

![image](https://cloud.githubusercontent.com/assets/7423187/9430977/2a5380cc-49ff-11e5-8934-853877470d52.png)

**Comment by halatmit** [comment]
Hmm ... Very interesting!!
Does it also crash on Developer 46?    Whatever is crashing for us started
happening at some point in the beta 45 series.
I'll play some more.




== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by halatmit** [comment]
@cpseager @jis @josmas @afmckinney @NeilFraser 

Wow!   It looks like you're right!    I didn't start looking for specific malformed quotations yet -- I suspect there are a lot.   But I removed all of the translation strings files from the App Inventor source, and the problem has gone away.   No to track sown an fix the actual offenders.

Thank you so much!  I can't imagine how I would have found this without your comment.  Do you want to post something on the web for other people who will be getting killed with the update to Chrome 45+?

**Comment by cpseager** [comment]
@halatmit Glad it helped, I suspected this error as I'd seen it before in our own use of Blockly.

We now always run the translation files through jsonlint.com as it tends to find any mistakes that will kill off Blockly

**Comment by halatmit** [comment]
Thanks! We'll try it.




== Hal Abelson
hal@mit.edu
Prof. of Comp. Sci. and Eng.
MIT Dept. of Elec. Eng. and Comp. Sci.
MIT Media Lab, Second Class Faculty

**Comment by jisqyv** [comment]
@cpseager @jis @josmas @afmckinney @NeilFraser I got things to work by removing the BOM at the beginning of the combined blockly.js. This is something we added years ago when we first added the Chinese translation. Without it Chrome would assume that the file was encoded as ASCII and fail when it found the Chinese characters (it would display them as ?'s or something equally useless). The BOM (U+FEFF) at the beginning of the file caused things to work correctly.

However now we are seeing weird errors, errors like "syntax error BloBlockly not defined". I believe something is broken in Chrome. However removing the BOM appears to have worked around the problem.

I have verified that things work in Firefox and Chrome stable without the BOM in place, so that will likely be the fix we go with.

-Jeff
